### PR TITLE
Fix #5409: Enable DApps support by default

### LINKED ISSUE #5409: Enable DApps support by default
Enable DApps for public builds

### PR DESCRIPTION
**Pull request by nuo-xu**
Enable DApps support for public builds

This pull request fixes #5409 

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`

## Reviewer Checklist:

- [ ] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [ ] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [ ] Adequate unit test coverage exists to prevent regressions.
- [ ] Adequate test plan exists for QA to validate (if applicable).
- [ ] Issue and pull request is assigned to a milestone (should happen at merge time).

### CODE REVIEW

File: Client/Frontend/Browser/UserScriptManager.swift [hunk 1]
```diff
@@ -459,14 +459,12 @@ class UserScriptManager {
...
-           tab?.isPrivate == false,
-           Preferences.Wallet.WalletType(rawValue: Preferences.Wallet.defaultWallet.value) == .brave {
-          $0.addUserScript(script)
-          if let providerJS = walletProviderJS {
-            $0.addUserScript(.init(source: providerJS, injectionTime: .atDocumentStart, forMainFrameOnly: true, in: .page))
-          }
+      if let script = walletProviderScript,
+         tab?.isPrivate == false,
+         Preferences.Wallet.WalletType(rawValue: Preferences.Wallet.defaultWallet.value) == .brave {
+        $0.addUserScript(script)
```
Comment (thypon) [inline Client/Frontend/Browser/UserScriptManager.swift:478]: shouldn't we use the `forMainFrameOnly` here too?

**Review by thypon — COMMENTED** [review]
Are we changing how the requester address for DApps is shown?

**Comment by StephenHeaps** [comment]
Once this PR merges (or as part of this PR), we should handle #5493

File: Client/Frontend/Browser/UserScriptManager.swift (hunk 1 above)
Comment (nuo-xu) [reply Client/Frontend/Browser/UserScriptManager.swift:478]: @thypon `walletProviderScript` is `forMainFrameOnly`. it is generated [here](https://github.com/brave/brave-ios/blob/6d5a843dbde8979212e18585eaea3284c21bf40c/Client/Frontend/Browser/UserScriptManager.swift#L399)

**Review by iccub — APPROVED** [review]

**Review by StephenHeaps — APPROVED** [review]

**Pull request merged**